docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,307 @@
+    ENCHANTMENT_COMPONENT_TYPES.register("example",
+        () -> DataComponentType.<ExampleEffectData>builder()
+            .persistent(ExampleEffectData.CODEC)
+            .build());
+```
+
+There are no inheritance requirements on the data held by an enchantment effect component.
+
+### `ConditionalEffect`
+Wrapping the type in `ConditionalEffect<?>` allows the enchantment effect component to take effect based on conditions informed by a [LootContext]. 
+
+Specifically, each `ConditionalEffect` contains another effect component, along with an `Optional<LootItemCondition>`. Since `LootItemContext` is a `Predicate<LootContext>`, it can be tested against a specified `LootContext` using `LootItemContext#test`.
+
+`ConditionalEffect` wraps this behavior, allowing one to simply call `ConditionalEffect#matches(LootContext context)` to determine if the effect should be allowed to run.
+
+Vanilla adds an additional helper method to further streamline the process of checking these conditions: `Enchantment#applyEffects()`. This method takes a `List<ConditionalEffect<T>>`, evaluates the conditions, and runs a `Consumer<T>` on each `T` contained by a `ConditionalEffect` whose condition was met. Since many vanilla enchantment effect components are defined as a `List<ConditionalEffect<?>>`, these can be directly plugged into the helper method like so:
+
+```java
+// `enchant` is an Enchantment instance.
+// `lootContext` is a LootContext instance.
+Enchantment.applyEffects(
+    enchant.getEffects(EnchantmentEffectComponents.KNOCKBACK), // Or whichever other List<ConditionalEffect<T>> you want
+    lootContext,
+    (effectData) -> // Use the effectData (in this example, an EnchantmentValueEffect) however you want.
+);
+```
+
+Registering a custom `ConditionalEffect`-wrapped enchantment effect component type can be done as follows:
+
+```java
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ConditionalEffect<ExampleData>>> EXAMPLE_CONDITIONAL_EFFECT =
+    ENCHANTMENT_COMPONENT_TYPES.register("example_conditional",
+        () -> DataComponentType.ConditionalEffect<ExampleData>builder()
+            .persistent(ConditionalEffect.codec(Unit.CODEC, LootContextParamSets.EMPTY))
+            .build());
+```
+The parameters to `ConditionalEffect.codec` are the codec for the generic `ConditionalEffect<T>`, followed by some `LootContextParamSets` entry.
+
+### Using Enchantment Effect Components
+
+Here is a full example using vanilla helper methods to work with a custom enchantment effect component.
+
+```java
+// Define an example data-bearing record.
+public record Increment(int value) {
+    public static final Codec<BoundEntity> CODEC = RecordCodecBuilder.create(instance ->
+            instance.group(
+                    Codec.INT.fieldOf("value").forGetter(Increment::value),
+            ).apply(instance, Increment::new)
+    );
+
+    public int add(int x) {
+        return value() + x;
+    }
+}
+```
+
+```java
```

Comment:
These do not need to be separate blocks. You can just say something like `// In some registry object class`